### ISSUE TITLE
tf.image.sobel_edges not supported in TPU

### ISSUE THREAD

**Issue by vinits5** [tensorflow/tensorflow#49509]
Hello,

I am training my deep neural network in TPU on google colab using tensorflow 2 and keras. I have added an L2 loss between the sobel edges of prediction and ground truth images. But, when I am calling `model.fit()` function, this `tf.image.sobel_edges` function raises errors. I believe that the function is not supported while compiling the graph in TPU. 

1. If I am correct then is there any alternate way to compute edges in TPU. 
2. And if I not correct then can you please point me to correct use of `tf.image.sobel_edges` function in TPU.
Thank you in advance.

**Comment by bhack** [comment]
Have you already tried with https://www.tensorflow.org/io/api_docs/python/tfio/experimental/filter/sobel ?

**Comment by tilakrayal** [comment]
@vinits5 ,

In order to expedite the trouble-shooting process, could you please provide the Tensorflow version, complete code and dataset to reproduce the issue reported here.

Thanks!

**Comment by vinits5** [comment]
This https://www.tensorflow.org/io/api_docs/python/tfio/experimental/filter/sobel ? worked. Thanks @bhack .
